openfl-tutorials/interactive_api/MXNet_landmarks/envoy/landmark_shard_descriptor.py

```diff
@@ -0,0 +1,169 @@
+        self.rank, self.worldsize = map(int, rank_worldsize.split(','))
+
+        self.data_folder = Path.cwd() / data_folder
+        self.download_data()
+
+        # Calculating data and target shapes
+        ds = self.get_dataset()
+        sample, target = ds[0]
+        self._sample_shape = [str(dim) for dim in sample.shape]
+        self._target_shape = [str(len(target.shape))]
+
+        if self._target_shape[0] != '1':
+            raise ValueError('Target has a wrong shape')
+
+    def process_data(self, name_csv_file) -> None:
+        """Process data from csv to numpy format and save it in the same folder."""
+        data_df = pd.read_csv(self.data_folder / name_csv_file)
+        data_df.fillna(method='ffill', inplace=True)
+        keypoints = data_df.drop('Image', axis=1)
+        cur_folder = self.data_folder.relative_to(Path.cwd())
+
+        for i in range(data_df.shape[0]):
+            img = data_df['Image'][i].split(' ')
+            img = np.array(['0' if x == '' else x for x in img], dtype='float32').reshape(96, 96)
+            np.save(str(cur_folder / f'img_{i}.npy'), img)
+            y = np.array(keypoints.iloc[i, :], dtype='float32')
+            np.save(str(cur_folder / f'keypoints_{i}.npy'), y)
+
+    def download_data(self) -> None:
+        """Download dataset from Kaggle."""
+        self.data_folder.mkdir(parents=True, exist_ok=True)
+
+        if not self.is_dataset_complete():
+            logger.info('Your dataset is absent or damaged. Downloading ... ')
+            api = KaggleApi()
+            api.authenticate()
+
+            if Path('data').exists():
+                shutil.rmtree('data')
+
+            api.competition_download_file(
+                'facial-keypoints-detection',
+                'training.zip', path=self.data_folder
+            )
+
+            with ZipFile(self.data_folder / 'training.zip', 'r') as zipobj:
+                zipobj.extractall(self.data_folder)
+
+            (self.data_folder / 'training.zip').unlink()
+
+            self.process_data('training.csv')
+            (self.data_folder / 'training.csv').unlink()
+            self.save_all_md5()
+
+    def get_dataset(self, dataset_type='train') -> LandmarkShardDataset:
+        """Return a shard dataset by type."""
+        return LandmarkShardDataset(
+            dataset_dir=self.data_folder,
+            rank=self.rank,
+            worldsize=self.worldsize
+        )
+
+    def calc_all_md5(self) -> Dict[str, str]:
+        """Calculate hash of all dataset."""
+        md5_dict = {}
+        for root in self.data_folder.glob('*.npy'):
+            md5_calc = md5()
+            rel_file = root.relative_to(self.data_folder)
+
+            with open(self.data_folder / rel_file, 'rb') as f:
+                for chunk in iter(lambda: f.read(4096), b''):
+                    md5_calc.update(chunk)
+                md5_dict[str(rel_file)] = md5_calc.hexdigest()
+        return md5_dict
```

Comment:
I'm not sure that it should be done in this PR, but I think that calculating single md5 hash enough.

Reply:
@dmitryagapov btw, we already have the similar functionality merged (also with Kaggle dataset stuff)
https://github.com/intel/openfl/blob/c1947fa236092cfb892d675a34f7c87dda92917c/openfl-tutorials/interactive_api/PyTorch_DogsCats_ViT/envoy/dogs_cats_shard_descriptor.py#L135